Altis_Life.Altis/core/functions/fn_startLoadout.sqf

```diff
@@ -6,23 +6,44 @@
     Description:
     Loads a custom loadout on player when he got a new life
 */
+private _side = call {
+    if (playerSide isEqualTo civilian) exitWith {"civ"};
+    if (playerSide isEqualTo west) exitWith {"cop"};
+    if (playerSide isEqualTo independent) exitWith {"med"};
+};
+
 private _level = call {
-    if (playerSide isEqualTo civilian) exitWith {
+    if (_side isEqualTo "civ") exitWith {
         if (life_is_arrested) then {
             "arrested"
         } else {
-            floor(random (count(missionConfigFile >> "Loadouts" >> str(playerSide))-1))
+            "random"
         };
     };
-    if (playerSide isEqualTo west) exitWith {
+    if (_side isEqualTo "cop") exitWith {
         FETCH_CONST(life_coplevel)
     };
-    if (playerSide isEqualTo independent) exitWith {
+    if (_side isEqualTo "med") exitWith {
         FETCH_CONST(life_mediclevel)
     };
 };
 
-player setUnitLoadout (missionConfigFile >> "Loadouts" >> str(playerSide) >> format["lvl_%1",_level]);
+if (_side isEqualTo "civ" && _level isEqualTo "random") then {
+    private _civLoadout = getUnitLoadout (missionConfigFile >> "Loadouts" >> "civ_level_random");
+    private _arr = [
+        "U_C_Poloshirt_blue",
+        "U_C_Poloshirt_burgundy",
+        "U_C_Poloshirt_stripped",
+        "U_C_Poloshirt_tricolour",
+        "U_C_Poloshirt_salmon",
+        "U_C_Poloshirt_redwhite",
+        "U_C_Commoner1_1"
+    ];
+	_civLoadout set [3, [(selectRandom _arr), []]];

```

Comment:
So this is a funny one, as you're setting the whole index of 3 as this info which means we could lose items here. We could be extra fancy and select into the array and set there.
```suggestion
(_civLoadout select 3) set [0,selectRandom _arr]
```

Reply:
I've tried that and it didn't work.